keras/applications/imagenet_utils.py

```diff
@@ -35,26 +45,27 @@ def preprocess_input(x, data_format=None, mode='caffe'):
         x *= 2.
         return x
 
+    vgg_mean = [103.939, 116.779, 123.68]
     if data_format == 'channels_first':
         if x.ndim == 3:
             # 'RGB'->'BGR'
             x = x[::-1, ...]
             # Zero-center by mean pixel
-            x[0, :, :] -= 103.939
-            x[1, :, :] -= 116.779
-            x[2, :, :] -= 123.68
+            if _IMAGENET_MEAN_CF is None:
+                _IMAGENET_MEAN_CF = np.array(vgg_mean, dtype=np.float32).reshape((3, 1, 1))
```

Comment:
should this be `np.float32` or `K.floatx()`?

Reply:
I'm not familiar with `K.floatx()` but after learning what it does I think it makes more sense.